lib/pf/pfmon/task/queue_stats.pm

```diff
@@ -0,0 +1,73 @@
+package pf::pfmon::task::queue_stats;
+
+=head1 NAME
+
+pf::pfmon::task::queue_stats - class for pfmon task queue counts
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::pfmon::task::queue_stats
+
+=cut
+
+use strict;
+use warnings;
+use pf::log;
+use pf::StatsD;
+use pf::pfqueue::stats;
+use Moose;
+extends qw(pf::pfmon::task);
+
+
+=head2 run
+
+Poll the queue counts and record them in statsd
+
+=cut
+
+sub run {
+    my ($self) = @_;
+    my $logger = get_logger;
+    $logger->debug("Polling counters from queues to record them in statsd");
+    my $statsd = pf::StatsD->new;
+
+    my $queue_counts = pf::pfqueue::stats->new->queue_counts;
+    for my $info (@{$queue_counts}) {
+        my $queue_name = $info->{name};
+        my $count = $info->{count};
+        $logger->debug("Setting queue count of $queue_name to $count in statsd");
+        $statsd->gauge("pfqueue.stats.queue_counts.$queue_name", $count, 1);
+    }
+}
+
```

Comment:
If you use ```pf::pfqueue::stats->stats_data```

We can get an even bigger picture of the state of the queue

```perl
my $stats_data = pf::pfqueue::stats->new->stats_data;
```
```$stats_data``` provided the following information

```perl
 {
          'queue_counts' => [
                              {
                                'count' => 0,
                                'name' => 'general'
                              },
                              {
                                'count' => 0,
                                'name' => 'pfdhcplistener'
                              },
                              {
                                'count' => 0,
                                'name' => 'pfdetect'
                              },
                              {
                                'count' => 0,
                                'name' => 'pfsnmp'
                              }
                            ],
          'miss_counters' => [
                               {
                                 'count' => '12194',
                                 'name' => 'api:echo',
                                 'queue' => 'general'
                               },
                               {
                                 'count' => '12194',
                                 'name' => 'api:sleep',
                                 'queue' => 'general'
                               },
                               {
                                 'count' => '12196',
                                 'name' => 'api:throw',
                                 'queue' => 'general'
                               }
                             ],
          'counters' => [
                          {
                            'count' => '0',
                            'name' => 'api:echo',
                            'queue' => 'general'
                          },
                          {
                            'count' => '0',
                            'name' => 'api:sleep',
                            'queue' => 'general'
                          },
                          {
                            'count' => '0',
                            'name' => 'api:throw',
                            'queue' => 'general'
                          }
                        ]
 };
```